Pet-Clinic-Frontend/src/pages/Registration.tsx

```diff
@@ -1,49 +1,116 @@
+            return false;
+        }
+        if (password.length < 6) {
+            setError("Password must be at least 6 characters");
+            return false;
+        }
+        return true;
+    }
 
-    const [email, setEmail] = useState(""); 
-    const [password, setPassword] = useState("") 
+    const checkEmailExists = (email: string) => {
+        //  API call. Not implemented in the backEnd yet.
 
-    const handleEmail = (e: React.ChangeEvent<HTMLInputElement>) => { 
-        setEmail(e.target.value); 
-        // setSubmitted(false); 
     }
-    
-    const handlePassword = (e: React.ChangeEvent<HTMLInputElement>) => { 
-        setPassword(e.target.value); 
-        // setSubmitted(false); 
+
+    const sendToBackend = async (email: string, password: string) => {
+        try {
+            const response = await fetch('http://localhost:8080/register', {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({ email, password }),
+            });
+            if (!response.ok) {
+                throw new Error('Network response was not ok');
+            }
+            const data = await response.json();
+            console.log('Success:', data);
+        } catch (error) {
+            console.error('Error:', error);
+            setError('Registration failed. Please try again.');
+        }
+    }
+
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        setError(null);
+
+        if (!validateForm()) {
+            return;
+        }
+
+/*         if (checkEmailExists(email)) {
+            setError("Email already exists");
+            return;
+        } not checkEmailExists implemented yet*/
+
+          // If all validations pass
+        setSubmitted(true);
+        sendToBackend(email, password);
+        setEmail("");
+        setPassword("");
     }
 
     return (
         <div>
-            <form className='register-form'> 
+            <form className='register-form' onSubmit={handleSubmit}>
                 <h3 className='rf-title'>Registration</h3>
-                <label className="label">Email</label> 
-                <input 
-                    onChange={handleEmail} 
+
+                {error && <div className="error-message">{error}</div>}
+                {submitted && <div className="success-message">From validation successful!</div>}
+
+                <label className="label">Email</label>
+                <input
+                    onChange={handleEmail}
                     className="input"
-                    value={email} 
-                    type="email"
-                /> 
-  
-                <label className="label">Password</label> 
-                <input 
-                    onChange={handlePassword} 
+                    value={email}
+                    type="text"
```

Comment:
why we cant use type email?

Reply:
I do not why but the response messages were not work with  type="email"